### PR TITLE
Make it possible to run examples from anywhere in the source tree

### PR DESCRIPTION
**Pull request by nchashch**
Now you can call "cargo run --example <example name>" from any directory,
not only from the root "Cargo.toml" directory.

### CODE REVIEW

**Comment by alteous** [comment]
It might be a better idea to embed the shader code into the program instead using `include_bytes!` or otherwise.

**Comment by alteous** [comment]
Unless the intention is that the shaders are overridable?

**Comment by vitvakatu** [comment]
@alteous we need flexible solution, to let users change their shaders easily. See #28

**Comment by kvark** [comment]
Thanks @nchashch !
We do need to change the way shaders are loaded - #28 
This PR solves a part of the issue, but sacrificing the initialization ergonomics. If we don't merge it, and focus on the proper solution, I don't think this is going to block anyone from playing with thee-rs, right?

**Comment by nchashch** [comment]
TL;DR: I think hardcoding shaders and keeping this `format!(..., env!(...))` solution for assets would be ok.

Yes, this is a bit of a crutch. Note that apart from shaders '.obj' and '.gltf' assets in current examples have this "run from only root Cargo.toml directory" problem. It is a minor problem, so I don't think solving only this problem (without any other benefit to the user) will be worth any kind of significant engineering effort.

I still think that being able to run examples from anywhere in the source tree is important, because it will save frustration to people tinkering with them. This "unwrap error" because you are in a wrong directory can be quite confusing for a person new to Rust.

Are there any plans for asset management? Three-rs has ergonomics focus and is targeted at making simple things, so having brutalist bare bones asset management - loading files specified by hard coded filenames might make sense to keep the API simple.

**Comment by kvark** [comment]
@nchashch 

> I think hardcoding shaders and keeping this format!(..., env!(...)) solution for assets would be ok.

I don't think this is what we want to end up with, but I'm fine with it being a temporary change.

> Are there any plans for asset management? 

Not at the moment, but we are open to the idea.

bors r+
